### PR TITLE
feat(core): platforms api

### PR DESCRIPTION
**Pull request by IT-MikeS**
Start of a more clean platforms api, for adding and setting platforms. Addresses #4243

### CODE REVIEW

File: core/src/runtime.ts [hunk 1]
```diff
@@ -22,17 +23,20 @@ export interface RegisteredPlugin {
...
-  const getPlatform = () => getPlatformId(win);
+  const defaultGetPlatform = () => getPlatformId(win);
+  const getPlatform = capPlatforms.currentPlatform.getPlatform || defaultGetPlatform;
 
-  const isNativePlatform = () => getPlatformId(win) !== 'web';
+  const defaultIsNativePlatform = () => capPlatforms.currentPlatform.name !== 'web';
+  const isNativePlatform = capPlatforms.currentPlatform.isNativePlatform || defaultIsNativePlatform;
 
-  const isPluginAvailable = (pluginName: string): boolean => {
+  const defaultIsPluginAvailable = (pluginName: string): boolean => {
```
Comment (adamdbradley) [inline core/src/runtime.ts:39]: Why does this have to get renamed?

File: core/src/runtime.ts [hunk 2]
```diff
@@ -22,17 +23,20 @@ export interface RegisteredPlugin {
...
   const cap: CapacitorInstance = win.Capacitor || ({} as any);
+  const capPlatforms: CapacitorPlatformsInstance = win.CapacitorPlatforms;
 
   const Plugins = (cap.Plugins = cap.Plugins || ({} as any));
 
   const webviewServerUrl =
     typeof win.WEBVIEW_SERVER_URL === 'string' ? win.WEBVIEW_SERVER_URL : '';
 
-  const getPlatform = () => getPlatformId(win);
+  const defaultGetPlatform = () => getPlatformId(win);
```
Comment (adamdbradley) [inline core/src/runtime.ts:33]: Not sure I follow why we need to have a `currentPlatform` and rename some of the apis to `default`

**Comment by adamdbradley** [comment]
Since this PR there has been a significant internal update to how the native bridge works. Overall I think it's a good update, but would rather not rename any of the public APIs we already have established. Would you be able to rebase from master and not rename any of the public APIs? Thanks

File: core/src/runtime.ts (hunk 1 above)
Comment (IT-MikeS) [reply core/src/runtime.ts:27]: The defaults are not renaming the consts that are exposed, they are just holding the default version of that function so in the case of a custom platform not overriding a piece then the runtime will have the default to fall back on

**Comment by IT-MikeS** [comment]
Did a rebase and a quick clean up

**Review by thomasvidas — APPROVED** [review]

**Comment by thomasvidas** [comment]
Looks great! If you need any help with getting https://github.com/capacitor-community/electron/ updated I'm happy to help. Just DM me on the community slack 😄

**Comment by IT-MikeS** [comment]
> Looks great! If you need any help with getting https://github.com/capacitor-community/electron/ updated I'm happy to help. Just DM me on the community slack 😄

Thanks! I'm pretty well done with the update but if anything comes up I'll be sure to take you up on that 😁

**Pull request merged**